include/sound/sof/header.h

```diff
@@ -153,6 +153,14 @@ struct sof_ipc_compound_hdr {
 	uint32_t count;		/**< count of 0 means end of compound sequence */
 }  __packed;
 
+struct sof_ipc_large_hdr {
+	struct sof_ipc_cmd_hdr hdr;
+	uint32_t cmd;
+	uint32_t count;
+	uint32_t id;
+	unsigned char data[];
+}  __packed;
+
```

Comment:
could we instead use the concept of compound_ipc, but make sure that there is a notion of start/stop and continuation between commands.

Reply:
I don't understand what is the point of having large message implementation if they fit our ipc frame? For e.g. fir filter coeffs are byte blobs that can be really long. If they are going to be transferred to dsp they have to be chopped and framed somehow. So they are not compound messages but 1 message that doesn't fit the transfer size.

I'm not totally sure how the kernel scheduling goes here: we are sending max size chunks in our "normal" ipc way, so I guess the kernel has time to schedule something else, but I guess other IPC at the same time to dsp might be difficult. We could disable large messages if there's stream playing?

Reply:
@plbossart @juimonen why can't we use a larger shared memory area? e.g reserve a chunk of system RAM for communication between DSP and CPU? Thus we can allocate few MB to allow large transfer with a single transfer. 

Basically, get rid of the 4K mailbox.